nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -227,10 +227,23 @@ in
       modulePackage = lib.mkOption {
         internal = true; # It is supposed to be selected automatically, but can be overridden by expert users.
         default = selectModulePackage cfgZfs.package;
+        apply = pkg: pkg.override { inherit (cfgZfs) allowNonLTSKernels; };
         type = lib.types.package;
         description = "Configured ZFS kernel module package.";
       };
 
+      allowNonLTSKernels = lib.mkEnableOption ''
+        usage of ZFS with non-LTS kernels that are likely to become
+        incompatible, preventing you from updating the system until you switch
+        to an LTS kernel.
+
+        Please see the evaluation warning of the ZFS package for more
+        information. If you have never seen such a warning, this option very
+        likely doesn't apply to you.
```

Comment:
I get that, yeah, but the thing is that it's hard to do this without duplicating the docstring; requiring updating both places.

Since this only applies to those who attempt to use such a configuration and will have seen the detailed warning, I think its fine this way.